src/no-side-effects.spec.ts

```diff
@@ -0,0 +1,129 @@
+        options: [{ excludedIdentifiers: ['assert', 'debug', 'log'] }],
+      },
+    ],
+    invalid: [
+      {
+        code: `import { strict as assert } from 'node:assert';
+               import debug from 'debug';
+               import Koa from 'koa';
+               import { v4 as uuid } from 'uuid';
+               import Router from '@koa/router';
+               import { StatusCodes } from 'http-status-codes';    
+               
+               const log = debug('report:event');
+              
+               const reportName = 'TEST_REPORT';
+
+               type records = Record;
+               
+               interface Report {
+                  reportName: string;
+                  report: [];
+               }
+               
+               function createS3ObjectKey(
+                from: string,
+                to: string,
+                bin: string,
+                reportId: string,
+                createdAt = new Date().toISOString(),
+              ): string {
+                // YYYY/MM/DD/HH
+                const prefix = formatUtc(createdAt, 'yyyy/MM/dd/HH');
+                return prefix;
+              }
+              assert(\`I'm a number, \${numberValue}\`);
+              const jsonSchemaValidator = new Ajv({ allErrors: true }).compile(schema);`,
+        errors: [{ message: 'No side effects can occur at the main module-level' }],
+        filename: 'src/side-effects.ts',
+        options: [{ excludedIdentifiers: ['assert', 'debug', 'log'] }],
+      },
+      {
+        code: `import { strict as assert } from 'node:assert';
+                import http from 'node:http';
+                import debug from 'debug';
+                
+                import { getConfiguration } from 'test';
+                
+                const log = debug('test:local');
+                
+                const unresolvedConfiguration = getConfiguration(root());
+                unresolvedConfiguration.api.root = 'build';
+                const configuration = await service(unresolvedConfiguration);
+                
+                assert.ok(configuration.requestHandler !== undefined, 'No request handler for root service');
+                assert.ok(unresolvedConfiguration.env['PORT'] !== undefined, 'PORT must be set');
+                
+                const server = http.createServer(logger(configuration.requestHandler, unresolvedConfiguration.name));
+                server.listen(Number.parseInt(unresolvedConfiguration.env['PORT'], 10), '0.0.0.0');
+                
+                // wait until the server is listening
+                await new Promise((resolve) => {
+                  server.on('listening', resolve);
+                });
+                
+                log(
+                  \`started \${unresolvedConfiguration.name} on http://localhost:\${unresolvedConfiguration.env['PORT']}, using node.js \${process.version}\`,
+                );`,
+        errors: [
```

Comment:
shall we split this test case (code) and maybe remove the valid statements as well so that it's easier to understand which statements triggger the errors?